apollo-router/src/json_ext.rs

```diff
@@ -528,7 +665,10 @@ impl Path {
                     } else if s == "@" {
                         PathElement::Flatten
                     } else {
-                        PathElement::Key(s.to_string())
+                        s.strip_prefix("... on ").map_or_else(
```

Comment:
Just a nit: that could be helpful to put `"... on "` in a constant somewhere to avoid typo and forget the trailing whitespace somewhere

Reply:
Good call. Pushed the quick change.